### PR TITLE
Switch to Dependabot for Python dependencies

### PR DESCRIPTION
**Pull request by foolip**
This is mainly because Dependabot treats a closed PR as "ignore this
version", why pyup does not:
https://github.com/web-platform-tests/wpt/pull/29159#issuecomment-851254537

### CODE REVIEW

**Comment by jgraham** [comment]
I remember proposing this before and there were some concerns, but I can't find the issue right now. I *think* it was just about the ability to pin py2-specific things, so it might not be relevant now.

In any case I think I'd prefer no having the weekly updates/limited open PRs setup. In particular for things like mozbase releases we're often taking the update to fix breakage so it makes sense to get the PR ASAP. Obviously it might be different if there were hundreds of deps, but in practice it's just a few that update often.

**Comment by jgraham** [comment]
I remember @gsnedders had opinions here too.

**Comment by foolip** [comment]
> In any case I think I'd prefer no having the weekly updates/limited open PRs setup

I've been a bit annoyed by how often we get hypothesis updates, but how about we make tools/wpt/ and tools/wptrunner/ daily? That's where the moz* dependencies are.

**Comment by jgraham** [comment]
Yeah, I'd be happy with having the pytest updates less often.

**Comment by foolip** [comment]
Regarding `open-pull-requests-limit`, the default is 5:
https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#open-pull-requests-limit

We could make it a number like 100 or 999999999 to make it more obvious we're going for no practical limit?

**Comment by jgraham** [comment]
YEah, some number higher than the number of deps we actually have seems fine.

**Comment by jgraham** [comment]
Re: pinning: https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates

**Comment by foolip** [comment]
Which part of that page covers pinning? Searching fails me...

**Comment by jgraham** [comment]
https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#ignore allows ignoring certain versions. I think you can do something like
```
ignore:
  - dependency-name: "pytest"
    versions: [">=6.0"]

**Comment by foolip** [comment]
Ah, great! I was looking for "pin".

Right now we aren't ignoring any versions, but once we land this we might have to do it to avoid certain upgrades.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,22 +1,47 @@
...
+  schedule:
+    interval: monthly
+  open-pull-requests-limit: 1000
+- package-ecosystem: pip
+  directory: tools/wpt/
+  schedule:
+    interval: daily
+  open-pull-requests-limit: 1000
+- package-ecosystem: pip
+  directory: tools/wptrunner/
```
[the remark this answers is not in the document]
Comment (foolip) [reply .github/dependabot.yml:44]: I don't know for sure. After reading https://github.community/t/dependabot-pip-what-filename-formats-does-it-look-for/160421/4 where a requirements/test.txt was picked up I thought maybe Dependabot is pretty liberal in finding requirements files. But https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#supported-package-ecosystems linked from that page suggests that it's not going to work.

So... maybe I should try this on my fork first.

[the remark this answers is not in the document]
Comment (foolip) [reply .github/dependabot.yml:44]: Looks like it will work: https://github.com/foolip/wpt/pull/20/files

**Review by jgraham — APPROVED** [review]

**Comment by foolip** [comment]
This didn't work quite like I'd hoped, we're getting separate PRs for each requirements file. I'll create a revert.

**Pull request merged**